### PR TITLE
Check for comments in collapsible ifs

### LINKED ISSUE #798: collapsible_if shouldn't apply if there is a comment before
In this example there is comment which applies to both branches in the `else` block.

```
        if entry.end > entry.base {
            prev = entry_; // Go to the next entry
        } else {
            // No usable memory in this entry. Remove it from the list.

            if prev == st.list {
                st.list = entry.next;
                prev = st.list;
            } else {
                (*prev).next = entry.next;
            }
        }
```

### PR DESCRIPTION
**Pull request by lukasstevens**
Fixes #798

### CODE REVIEW

**Comment by lukasstevens** [comment]
I should probably also add the possibility of `/* */` multi-line comments.

File: clippy_lints/src/collapsible_if.rs [hunk 1]
```diff
@@ -112,9 +112,17 @@ fn check_if(cx: &EarlyContext<'_>, expr: &ast::Expr) {
     }
 }
 
+fn block_starts_with_comment(cx: &EarlyContext<'_>, expr: &ast::Block) -> bool {
+    // The zeroth character in the trimmed block text is "{", which marks the beginning of the block.
+    // Therefore, we check if the first string after that is a comment, i.e. starts with //.
+    let trimmed_block_text = snippet_block(cx, expr.span, "..").trim_left().to_owned();
+    trimmed_block_text[1..trimmed_block_text.len()].trim_left().starts_with("//")
+}
+
```
Comment (matthiaskrgr) [inline clippy_lints/src/collapsible_if.rs]: Hmm, this looks like it would fail for comments like `/* this */`

Comment (matthiaskrgr) [reply clippy_lints/src/collapsible_if.rs]: Ah nevermind, I didn't see your comment.. :/

**Comment by flip1995** [comment]
Thanks! The incorrect indentation on the suggestion is unrelated, see #2974

**Pull request merged**